starknet_py/serialization/data_serializers/uint_serializer.py

```diff
@@ -0,0 +1,75 @@
+from dataclasses import dataclass
+from typing import Generator, TypedDict, Union
+
+from starknet_py.cairo.felt import uint256_range_check
+from starknet_py.serialization import CairoDataSerializer
+from starknet_py.serialization._context import (
+    Context,
+    DeserializationContext,
+    SerializationContext,
+)
+
+U128_UPPER_BOUND = 2**128
```

Comment:
Here it does not matter but you could always do `1 << 128` which is much faster. (again here it does not matter 🗿)

Reply:
I might be confused: should uint128 upper bound be just `2 ** 128` or `(2 ** 128) - 1`?

Reply:
Depends on how it is used, but the name suggests it should be `-1`. Good catch, let's double check on that.

Reply:
It is used like this `0 <= value < U128_UPPER_BOUND`, so the value is fine

Reply:
I would keep it as it is because this form is much more readable